src/naive/date.rs

```diff
@@ -768,11 +768,11 @@ impl NaiveDate {
             return Some(self);
         }
 
-        i64::try_from(days.0).ok().and_then(|d| self.add_days(-d))
+        i32::try_from(days.0).ok().and_then(|d| self.add_days(-d))
     }
 
-    fn add_days(self, days: i64) -> Option<Self> {
-        let secs = days.checked_mul(86400)?; // 86400 seconds in one day
+    fn add_days(self, days: i32) -> Option<Self> {
```

Comment:
Why is this useful/necessary?

Reply:
The type of `NaiveDate` is an `i32`, and we don't even use all the bits of it for storing the date.
So any value larger than `i32` doesn't make sense.

To me the question is more: why did it accept an `i64` in the first place? Because it was part of adding a 64-bit duration.

Reply:
Or do you mean: why include these commits? It avoids `Duration::second`, which is not const in time 0.1.
And they were just a nice simplification in my opinion.

Reply:
No, I was asking the former question. Sounds good.